File: app/src/main/java/org/mifos/selfserviceapp/injection/module/ActivityModule.java
package org.mifos.selfserviceapp.injection.module;

import android.app.Activity;
import android.content.Context;

import org.mifos.selfserviceapp.injection.ActivityContext;

import dagger.Module;
import dagger.Provides;

/**
 * @author ishan
 * @since 08/07/16
 */
@Module
public class ActivityModule {

    private Activity activity;

    public ActivityModule(Activity activity) {
        this.activity = activity;
    }

    @Provides
    Activity providesActivity() {
        return activity;
    }

    @Provides
    @ActivityContext
    Context providesContext() {
        return activity;
    }

}

File: app/src/main/java/org/mifos/mobile/injection/module/ActivityModule.kt
package org.mifos.mobile.injection.module

import android.app.Activity
import android.content.Context
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import org.mifos.mobile.injection.ActivityContext

/**
 * @author ishan
 * @since 08/07/16
 */
@Module
@InstallIn(SingletonComponent::class)
class ActivityModule(private val activity: Activity? = null) {
    @Provides
    fun providesActivity(): Activity {
        return activity!!
    }

    @Provides
    @ActivityContext
    fun providesContext(): Context {
        return activity!!
    }
}
